Automattic/woocommerce-payments
# Change payment method title for In Person Card Payment

Fixes #1757 

#### Changes proposed in this Pull Request

The orders paid for by in-person / card present payments show "via WooCommerce Payments" in the order details page.
 
<img width="1300" alt="Screen Shot 2021-11-04 at 18 03 40" src="https://user-images.githubusercontent.com/1553182/140385681-550f650b-5c28-4c09-b6ab-0113cd702c58.png">

This PR replaces the text "via WooCommerce Payments" with "via In-Person Card Payment" for this type of orders.

<img width="1300" alt="Screen Shot 2021-11-04 at 18 09 10" src="https://user-images.githubusercontent.com/1553182/140386806-063bc66a-8236-4c6a-957a-ee9125e52946.png">

#### Testing instructions
(inspired by testing instructions in this [PR](https://github.com/Automattic/woocommerce-payments/pull/1689))

* Create an order with Cash on Delivery as payment method. You can also create the order in wp-admin. Note the order id.
* In Stripe dashboard, create a new payment with manual capture (More options > Capture funds later). Note the intent ID
*[ Change the permission_callback on the new endpoint](https://github.com/Automattic/woocommerce-payments/blob/236e39b4d4f4b7efdd9e36121b19a6861c6ded53/includes/admin/class-wc-rest-payments-orders-controller.php#L61) to '__return_true'. You need this to be able to make the request to the endpoint in next step. 
```diff
-  'permission_callback' => [ $this, 'check_permission' ],
+  'permission_callback' => '__return_true',
```
* Capture the payment making a request to the `capture_terminal_payment` endpoint. For instance, `curl -i -X POST <YOUR_SITE_URL>/wp-json/wc/v3/payments/orders/<ORDER_ID>/capture_terminal_payment --data "payment_intent_id=<INTENT_ID>"`
* The request should succeed. You can go to the details page of the order and check that the payment information is updated, as well as the "via In-Person Card Payment" text (please check screenshot above for reference)

-------------------


**Post merge**
